modules/javafx.controls/src/test/java/test/javafx/scene/control/ComboBoxTest.java

```diff
@@ -285,6 +285,69 @@ public Node getDisplayNode() {
         assertNull(comboBox.getSelectionModel());
     }
 
+    @Test public void testNullSelectionModelDoesNotThrowNPEInSkinOnValueChange() {
+        ObservableList<String> items = FXCollections.observableArrayList("ITEM1", "ITEM2");
+
+        ListCell<String> buttonCell = new ListCell<>() {
+            @Override
+            protected void updateItem(String item, boolean empty) {
+                super.updateItem(item, empty);
+                setText(item);
+            }
+        };
+        comboBox.setButtonCell(buttonCell);
+        comboBox.setItems(items);
+        comboBox.setSelectionModel(null);
+
+        comboBox.setValue(items.get(1));
+
+        assertEquals(items.get(1), comboBox.getButtonCell().getText());
+    }
+
+    @Test public void testNullSelectionModelDoesNotThrowNPEInSkinOnLayout() {
+        ObservableList<String> items = FXCollections.observableArrayList("ITEM1", "ITEM2");
+
+        ListCell<String> buttonCell = new ListCell<>() {
+            @Override
+            protected void updateItem(String item, boolean empty) {
+                super.updateItem(item, empty);
+                setText(item);
+            }
+        };
+        comboBox.setButtonCell(buttonCell);
+        comboBox.setItems(items);
+
+        comboBox.setValue(items.get(1));
+        comboBox.setSelectionModel(null);
+
+        comboBox.layout();
+
```

Comment:
KISS again :)

- buttonCell is unrelated
- setting the value is a not immediately obvious trigger of the NPE
- the test relies on the fact that the combo's skin is installed in setup - when installing it later, we'd run into a NPE at a different location in code

but then, this might be something to remember for a future bug/fix, if we agree on not including the list/combo sync into this bug

Reply:
ButtonCell strikes again. 😄 I will remove it.
Hm, this sounds like a follow-up then. Should I test this and create one?

Reply:
the buttonCell state is simply irrelevant here, what's striking - you - again is a test without final assert :) Same as toggling editability.